File: src/main/java/org/akoshterek/backgammon/move/MoveType.java
package org.akoshterek.backgammon.move;

/**
 * @author Alex
 *         date 20.07.2015.
 */
public enum MoveType {
    MOVE_INVALID (-1),
    MOVE_GAMEINFO(0),
    MOVE_NORMAL(1),
    MOVE_SETDICE(7);

    private final int value;

    private MoveType(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }
}

File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/move/MoveType.scala
package org.akoshterek.backgammon.move

/**
  * @author Alex
  *         date 20.07.2015.
  */
sealed trait MoveType {
    val value: Int
}

object MoveType {
    case object MOVE_INVALID extends MoveType {val value: Int = -1}
    case object MOVE_GAMEINFO extends MoveType {val value: Int = 0}
    case object MOVE_NORMAL extends MoveType {val value: Int = 1}
    case object MOVE_SETBOARD extends MoveType {val value: Int = 6}
    case object MOVE_SETDICE extends MoveType {val value: Int = 7}
}
